### PR TITLE
 feature(Rubric Payroll)

### PR DESCRIPTION
**Pull request by lomamech**
feature(Rubric Payroll)  
- Implement modul Rubric Payroll Management
- Implement API /admin/rubrics Manangement
- Fix some translate key in french repository
- Implement test Integration for Management Rubric
- Implement test E2E Rubric Payroll Management

reference #2292

### CODE REVIEW

File: client/src/i18n/en/rubric.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{
+  "RUBRIC" : {
+    "ADD_RUBRIC" : "Add a Rubric",
+    "ALL_RUBRICS" : "All Rubrics",
+    "CREATED" : "Rubric created with success",
```
Comment (DedrickEnc) [inline client/src/i18n/en/rubric.json:5]: Keys like ```CREATED```, ```DELETED```, ```UPDATED``` can be moved to ```FORM.INFO``` in `form.json` or they can be removed and choose other keys in ```FORM.INFO``` to replace them.

File: client/src/i18n/fr/depot.json [hunk 2]
```diff
@@ -4,7 +4,7 @@
     "ALL_DEPOTS" : "Tous les dépôts",
     "CREATED" : "Dépôt crée avec succès",
     "CHANGE" : "Changer Dépôt",
-    "DELETED" : "Dépôt supprimer avec succès",
+    "DELETED" : "Dépôt supprimé avec succès",
```
Comment (DedrickEnc) [inline client/src/i18n/fr/depot.json:7]: `DELETED`, `CREATED`, are notification keys you can use keys in ```FORM.INFO``` in the `form.json` file or add new one

File: client/src/i18n/fr/form.json [hunk 3]
```diff
@@ -352,6 +353,10 @@
             "IS_REPORT": "Est ce un repport ?",
             "IS_TITLE": "Est ce un titre?",
             "IS_WAREHOUSE": "Entreprot ?",
+            "IS_ADVANCE"     : "Est-ce une avance ?",
+            "IS_DISCOUNT"    : "Est-ce une réduction ?",
+            "IS_PERCENT"     : "Est-ce en(%) ?",
+            "IS_SOCIAL_CARE" : "Est une charge sociale ?",
```
Comment (DedrickEnc) [inline client/src/i18n/fr/form.json:359]: I think the key should be `IS_SOCIAL_CHARGE`

File: client/src/i18n/fr/grade.json [hunk 4]
```diff
@@ -4,7 +4,7 @@
     "ALL_GRADES" : "Tous les grades",
     "CREATED" : "Grade crée avec succès",
     "CHANGE" : "Changer Grade",
-    "DELETED" : "Grade supprimer avec succès",
+    "DELETED" : "Grade supprimé avec succès",
```
Comment (DedrickEnc) [inline client/src/i18n/fr/grade.json:7]: The same comment than above

File: client/src/i18n/fr/rubric.json [hunk 5]
```diff
@@ -0,0 +1,18 @@
+{
+  "RUBRIC" : {
+    "ADD_RUBRIC" : "Ajouter une rubrique",
+    "ALL_RUBRICS" : "Toutes les rubriques",
+    "CREATED" : "Rubrique créée avec succès",
```
Comment (DedrickEnc) [inline client/src/i18n/fr/rubric.json:5]: Same comment than above

File: client/src/i18n/fr/tree.json [hunk 6]
```diff
@@ -67,7 +67,7 @@
     "REFERENCE_GROUP":"Groupe de référence",
     "REGISTER_SUPPLIER":"Fournisseurs",
     "ROOT":"Racine",
-    "RUBRIC_MANAGEMENT":"Gestion Rubriques",
+    "RUBRIC_MANAGEMENT":"Gestion des Rubriques",
```
Comment (DedrickEnc) [inline client/src/i18n/fr/tree.json:70]: Gestion Rubrique des Paie is more clear what do you think?

**Review by DedrickEnc — CHANGES_REQUESTED** [review]
@lomamech 

I am glad to see that we begin to tackle payroll module. After our first installation attempt and the difficulties we faced, and after reading the issue #2292, here are some points to consider in order to implement payroll module more efficiently : 

- The table **contributions** should be called **deductions** because it should contain every element to deduct before applying any kind of taxes (**avance sur salaire**, **INSS**, **Pret divers** are deductions)

- The table **taxes** should be called **contributions**, the inspirations come form ***payNet enterprise*** which is in charge of paying IMA staff. (**IPR**, **ONEM**, **INPP** are contributions)

- The table **rubrics** should be called **pay_elements** as contributions, deductions and pay_elements are rubrics.

- Each of them (contributions, pay_elements, deductions) must have a **four and a six account** (one of feedback we received from Tshikaji and approved by many accountants).

- Each employee must have a basic salary if `NULL` the grade basic salary will be used in place.

here are options, for implementations : 

**options 1 (table SQL statements)**

```
CREATE TABLE pay_elements (
  id int(3) unsigned NOT NULL AUTO_INCREMENT,
  label VARCHAR(30),
  is_taxable tinyint(1) DEFAULT 0,
  `is_percent` tinyint(1) DEFAULT 0,
  `value` float DEFAULT 0,
  PRIMARY KEY (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8;
``` 

```
CREATE TABLE contributions (
  id int(3) unsigned NOT NULL AUTO_INCREMENT,
  label VARCHAR(30),
  is_employee tinyint(1) DEFAULT 0,
  `is_percent` tinyint(1) DEFAULT 0,
  `value` float DEFAULT 0,
  PRIMARY KEY (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8;
``` 

```
CREATE TABLE deductions (
  id int(3) unsigned NOT NULL AUTO_INCREMENT,
  label VARCHAR(30),
  is_employee tinyint(1) DEFAULT 0,
  `is_percent` tinyint(1) DEFAULT 0,
  `value` float DEFAULT 0,
  PRIMARY KEY (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8;
```

**Options 2**

Creating one table with all rubrics and put flags to distinguish them (contributions, pay_elements, and deductions)

This is a good chance for us to revisit these modules and clean them instead of just putting them in BHIMA 2 without changes.

@IMA-WorldHealth/bhima-core let me know what do you think

**Comment by DedrickEnc** [comment]
After this deep change, The review can continue

**Comment by lomamech** [comment]
@DedrickEnc  @jniles 

In the optics that must be added, the four account and the six account at the table "rubric", it seems good to me to gather all these different elements in one table, so I want to block all the PR in course (Rubric, Tax and Contribution) and merge them into a single Table payroll rubric

**Pull request closed without merging**